chaosaws/cloudwatch/probes.py

```diff
@@ -97,39 +109,47 @@ def get_metric_statistics(namespace: str, metric_name: str,
         )
 
 
-def get_metric_data(namespace: str, metric_name: str, dimension_name: str,
-                    dimension_value: str, statistic: str = None,
+def get_metric_data(namespace: str, metric_name: str,
+                    dimension_name: str = None, dimension_value: str = None,
+                    dimensions: List[Dict[str, Any]] = None, statistic: str = None,
                     duration: int = 300, period: int = 60, offset: int = 0,
                     unit: str = None, configuration: Configuration = None,
                     secrets: Secrets = None) -> float:
-    """Gets metric data for a given metric in a given time period. This method
-    allows for more data to be retrieved than get_metric_statistics
-
-    :params
-        namespace: The AWS metric namespace
-        metric_name: The name of the metric to pull data for
-        dimension_name: The name of the dimension to search for
-        dimension_value: The value to be used for searching the dimension
-        unit: The type of unit desired to be collected
-        statistic: The type of data to return.
+    """
+    Gets metric data for a given metric in a given time period. This method
+    allows for more data to be retrieved than get_metric_statistics.
+
+        :param namespace: The AWS metric namespace
+        :param metric_name: The name of the metric to pull data for
+        :One of:
+            :params dimension_name, dimension_value: Required to search for ONE dimension
+            :param dimensions: Required to search for dimensions combinations 
+            Are expected as a list of dictionary objects: [ {‘Name’: ‘DimName1’, ‘Value’: ‘Value1’}, {‘Name’: ‘DimName2’, ‘Value’: ‘Value2’}, … ]
+        :param unit: The type of unit desired to be collected
+        :param statistic: The type of data to return
             One of: Average, Sum, Minimum, Maximum, SampleCount
-        period: The window in which to pull datapoints for
-        offset: The time (seconds) to offset the endtime (from now)
-        duration: The time (seconds) to set the start time (from now)
+        :param period: The window in which to pull datapoints for
+        :param offset: The time (seconds) to offset the endtime (from now)
+        :param duration: The time (seconds) to set the start time (from now)
+
+    More information about input parameters are available in the documentation
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudwatch.html#CloudWatch.Client.get_metric_data
     """
     start_time = datetime.utcnow() - timedelta(seconds=duration)
     end_time = datetime.utcnow() - timedelta(seconds=offset)
+    
```

Comment:
Could we please have the WARNING statement somewhere indicating to users that the `dimension_name` and `dimension_value` option is now not the supported boto way, that folks should aim to change to passing us a list of `dimensions`

Reply:
Upon more thinking about this, I think it's fine that we support both. I'm not opposed to not warn anyone here.